airflow/utils/types.py

```diff
@@ -54,7 +54,7 @@ class AttributeRemoved:
     """
 
     def __getattr__(self, item):
-        raise RuntimeError("Attribute was removed on serialization and must be set again.")
+        raise RuntimeError(f"Attribute was removed on serialization and must be set again: {item}.")
```

Comment:
i think the reason i did not do this before is because when you hit this, item will be the thing on DAG requested, not `dag` for example.

e.g. it would fire if you did 
`ti.task.dag.task_dict` and the error would be like
```
Attribute was removed on serialization and must be set again: task_dict
```
but the thing that was actually removed is `dag`
not that this isn't a fine thing to do but that was the thought

Reply:
Yes. Noticed that - actually bes thing would be to get the name of removed atteribute. But it at least gace a clue - because without the context it's meaningless error.

Reply:
Yeah it's not completely meaningless.  You would have this statement `Attribute was removed on serialization and must be set again` and then to answer the question "which attribute" you have the traceback.  Referencing the wrong attr might be worse than not referencing any.  That's where i landed anyway.  But no objection from me if you think this is better.